core/src/test/java/org/cru/contentscoring/core/service/impl/ContentScoreUpdateServiceImplTest.java

```diff
@@ -150,33 +155,128 @@ public void testSetContentScoreUpdateDateRetry() throws RepositoryException {
         Page page = mockPage(pagePath, vanityPath, "https://vanity.com/content/test/us/en/" + vanityPath);
         when(page.getVanityUrl()).thenReturn(vanityPath);
 
-        String vanityUrl = updateService.getPageUrl(page);
+        String vanityUrl = updateService.getPageUrl(page, true);
         assertThat(vanityUrl, is(equalTo("https://vanity.com/" + vanityPath)));
     }
 
     @Test
-    public void testGetPageUrl() {
+    public void testGetPageUrl() throws Exception {
         String pagePath = "/content/test/us/en/page-path";
         Page page = mockPage(pagePath, pagePath, "https://page.com" + pagePath);
 
-        String pageUrl = updateService.getPageUrl(page);
-        assertThat(pageUrl, is(equalTo("https://page.com" + pagePath + ".html")));
+        String pageUrl = updateService.getPageUrl(page, false);
+        assertThat(pageUrl, is(equalTo("https://page.com" + pagePath + HTML_EXTENSION)));
     }
 
-    private Page mockPage(final String pagePath, final String externalizerPath, final String externalLink) {
+    @Test
+    public void testDoNotSkipPageUrl() throws Exception {
+        String vanityPath = "/vanity-url";
+        String pagePath = "/content/test/us/en/page-path";
+        String prefix = "https://vanity.com/content/test/us/en";
+
+        Page page = mockPage(pagePath, vanityPath, prefix + vanityPath);
+        when(page.getVanityUrl()).thenReturn(vanityPath);
+
+        String vanityUrl = updateService.getPageUrl(page, false);
+        assertThat(vanityUrl, is(equalTo(prefix + vanityPath + HTML_EXTENSION)));
+    }
+
+    @Test
+    public void testSkipVanityUrl() throws Exception {
+        String pagePath = "/content/test/us/en/page-path";
+        Page page = mockPage(pagePath, pagePath, "https://page.com" + pagePath);
+
+        String pageUrl = updateService.getPageUrl(page, true);
+        assertThat(pageUrl, is(nullValue()));
+    }
+
+    @Test
+    public void testPageWithVanityUrlSendsBothUrls() throws Exception {
+        initializeQueue();
+        String vanityPath = "/vanity-url";
+        String pagePath = "/content/test/us/en/page-path";
+        String prefix = "https://vanity.com/content/test/us/en";
+
+        Page page = mockPage(pagePath, vanityPath, prefix + vanityPath);
+        when(page.getVanityUrl()).thenReturn(vanityPath);
+
+        updateService.updateContentScore(page);
+        List<ContentScoreUpdateRequest> pending =
+            ContentScoreUpdateServiceImpl.internalQueueManager.getPendingBatches();
+
+        assertThat(pending, is(not(nullValue())));
+        assertThat(pending.size(), is(equalTo(2)));
+
+        int correctPaths = 0;
+
+        for (ContentScoreUpdateRequest request : pending) {
+            if (request.getUri().equals("https://vanity.com" + vanityPath)
+                || request.getUri().equals(prefix + vanityPath + HTML_EXTENSION)) {
+                correctPaths++;
+            }
+        }
+
+        assertThat(correctPaths, is(equalTo(2)));
+    }
```

Comment:
So this test sends:
 * `https://vanity.com/content/test/us/en/vanity-url.html`
 * `https://vanity.com/vanity-url`

The latter makes sense to me, but I thought the former should have been `https://vanity.com/content/test/us/en/page-path.html`. Am I wrong, and we actually want vanity-url.html? (I am admittedly not very knowledgeable about how vanity urls work.)

Reply:
So, it has to do with how the externalizer works. I ran tests through again to see it was doing what I had thought when I first wrote these tests:

```
No Vanity URL: http://local.cru.org:4503/content/cru/us/en/how-to-know-god.html
Vanity URL: 
  http://local.cru.org:4503/content/cru/us/en/testing.html,
  http://local.cru.org:4503/testing 
(vanityUrl = /content/cru/us/en/testing, 
 path = /content/cru/us/en/how-to-know-god)
```

I need to make sure that it is correct to send it that way, rather than sending the path, but the test is accurate to what the code is doing.

Reply:
ok, once you make sure that is correct, then this PR is good to merge.